instagram/src/components/Authentication/Authenticate.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+
+const Authenticate = (App, Login) =>
+class extends React.Component {
+    constructor(props){
+        super(props);
+        this.state = {
+            loggedIn: false,
+            username: '',
+            password: ''
+        }
+
+        this.login = this.login.bind(this);
+        this.inputHandler = this.inputHandler.bind(this);
+        this.logOut = this.logOut.bind(this);
+    }
+
+    login(event){
+        event.preventDefault();
+        console.log(window.localStorage.getItem('loggedIn'));
+        if(this.state.username && this.state.password){
+            window.localStorage.setItem('loggedIn', 'true');
+            window.localStorage.setItem('username', this.state.username);
+            this.setState({loggedIn: true});
+        }
+        
+      }
+
+      logOut(){
+            window.localStorage.removeItem('loggedIn');
+      }
+
+      inputHandler(event){
+          if(event.target.name === 'username'){
+              this.setState({username: event.target.value});
+          }
+          else if(event.target.name === 'password'){
+              this.setState({password: event.target.value});
+          }
+
+      }
+
+       componentDidMount(){
+           console.log(window.localStorage.getItem('loggedIn'));
+        let loggedIn = window.localStorage.getItem('loggedIn');
+        if (loggedIn){
+            let usernameCheck = window.localStorage.getItem('username');
+            this.setState({loggedIn: true, currentUser: usernameCheck});
+        }
+        else {
+            this.setState({loggedIn: false, currentUser: ''});
+        } 
+        
+            
+        
+    }  
+
+    render(){
+        return (
+        <div>
+        {window.localStorage.getItem('loggedIn') ? <App username={this.state.username} logOut={this.logOut} /> : <Login login={this.login} onKeyDown={this.inputHandler}/>}
```

Comment:
use this.state for your conditional, that way when you update state to reflect being logged in, you will automatically switch to rendering App.

Also, be sure to spread props onto App. Like this: `<App {...props} />`